CMakeLists.txt

```diff
@@ -14,6 +16,35 @@ if (NOT DEFINED LLVM_SPIRV_BUILD_EXTERNAL)
   endif(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
 endif (NOT DEFINED LLVM_SPIRV_BUILD_EXTERNAL)
 
+# Download spirv.hpp from the official SPIRV-Headers repository.
+# One can skip this step by manually setting
+# LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR path.
+if (NOT DEFINED LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR)
+  if (LLVM_SPIRV_BUILD_EXTERNAL)
+    set(LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR
+      "${CMAKE_CURRENT_BINARY_DIR}/SPIRV-Headers")
+  else() # In-tree build
+    set(LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR
+      "${CMAKE_SOURCE_DIR}/projects")
```

Comment:
Even though `llvm/projects` directory is listed in [`.gitignore`](https://github.com/llvm/llvm-project/blob/main/llvm/.gitignore) in the upstream LLVM, I'm not sure that we should update it every time we do the build - I suggest downloading headers only to `build` area regardless of in-tree/out-of-tree setup

Reply:
Done.